File: Haskell/5_calculo_de_media_ponderada.hs
main = do
    putStrLn "Digite nota 1:"
    inputNota1 <- getLine
    putStrLn "Digite nota 2:"
    inputNota2 <- getLine
    putStrLn "Digite nota 3:"
    inputNota3 <- getLine
    
    putStrLn "Digite peso 1:"
    inputPeso1 <- getLine
    putStrLn "Digite peso 2:"
    inputPeso2 <- getLine
    putStrLn "Digite peso 3:"
    inputPeso3 <- getLine
    
    let nota1 = read inputNota1 :: Double
        nota2 = read inputNota2 :: Double
        nota3 = read inputNota3 :: Double
        peso1 = read inputPeso1 :: Double
        peso2 = read inputPeso2 :: Double
        peso3 = read inputPeso3 :: Double
        
        somaPesos = peso1 + peso2 + peso3
        mediaPonderada = (nota1 * peso1 + nota2 * peso2 + nota3 * peso3) / somaPesos
        
    putStrLn $ "Média ponderada = " ++ show mediaPonderada

File: Java/5_calculo_de_media_ponderada.java
import java.util.Scanner;
import java.text.DecimalFormat;

class MediaPonderada {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        DecimalFormat formato = new DecimalFormat("#.##");

        System.out.print("Digite a nota 1:");
        float nota1 = scanner.nextInt();

        System.out.print("Digite a nota 2:");
        float nota2 = scanner.nextInt();

        System.out.print("Digite a nota 3:");
        float nota3 = scanner.nextInt();

        System.out.print("Digite o peso 1:");
        float peso1 = scanner.nextInt();

        System.out.print("Digite o peso 2:");
        float peso2 = scanner.nextInt();

        System.out.print("Digite o peso 3:");
        float peso3 = scanner.nextInt();

        float mediaPonderada = (nota1 * peso1 + nota2 * peso2 + nota3 * peso3) / (peso1 + peso2 + peso3);
        System.out.println("\nA média ponderada é: " + formato.format(mediaPonderada));
    }
}
